karma.conf.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+module.exports = function(config) {
+
+  config.set({
+    // base path that will be used to resolve all patterns (eg. files, exclude)
+    basePath: '',
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: ['mocha', 'chai'],
+
+    // list of files / patterns to load in the browser
+    files: [
+      'tests/*.spec.js'
+    ],
+
+    // list of files to exclude
+    exclude: ['bin/*'],
+    preprocessors: {
+      'tests/*.js': ['webpack', 'babel', 'sourcemap']
+    },
+
+    webpack: {
+      //mode: 'production',
+      //mode: 'development',
+      devtool: 'inline-source-map',
+      module: {
+        rules: [
+          {
+            test: /\.js$/,
+            exclude: [
+              /bin/,
+              /node_modules\/(?!jsonld|crypto-ld)/
+            ],
+            use: {
+              loader: 'babel-loader',
+              options: {
+                presets: ['@babel/preset-env'],
+                plugins: [
+                  '@babel/plugin-transform-modules-commonjs',
+                  '@babel/plugin-transform-runtime',
+                  "@babel/plugin-proposal-object-rest-spread"
+                ]
+              }
+            }
+          }
+        ]
+      },
+      node: {
+        Buffer: false,
+        process: false,
+        // this option means that a polyfill is used in web browsers
+        // providing node's crypto interface in the browser
+        // https://webpack.js.org/configuration/node/
+        crypto: true,
```

Comment:
@davidlehn can you look here and say whether or not this is doing the right thing? Seems like it's doing the opposite of what we want... but it's under a `node` section rather than a browser or exports section, so I find it confusing.

Reply:
just so you know bitcore-lib requires node's crypto module. it looks like vc-js is node crypto first .

Lib is here:

https://github.com/bitpay/bitcore/tree/v8.0.0/packages/bitcore-lib

Reply:
or maybe any crypto polyfil will do?

Reply:
@dlongley @davidlehn check the latest here:
using ProvidePlugin and an alias so we have one consistent crypto:
https://github.com/digitalbazaar/vc-js/pull/4/commits/95e7d023b88341d073324ca1633285d928343e9c

Reply:
@aljones15, my understanding is that this `crypto` lib is for providing a module that implements the *node.js* crypto API on the browser (the only thing that's confusing me is that it is under a `node` section, rather than `exports` or `browser` as I'm used to, so maybe that's incorrect). We don't need or want any node.js crypto API/module in the browser at all, not even for the bitcore stuff, which we need to split out into a separate library so we can avoid this issue better in the future. I'd expect the bitcore stuff to be non-functional and that's ok.

@davidlehn -- really need your thoughts here.

Reply:
@dlongley the alias actually that points to isomorphic-crypto actually does fit the needs for bitcore-lib (no errors!) that said the tests as they stand actually pass even if bitcore-lib can not find crypto which suggests that whatever needs bitcore (looks like jsonld-signatures) might not be used in these tests. even stranger though:

https://github.com/bitpay/bitcore/tree/v8.0.0/packages/bitcore-lib:

>To build a bitcore-lib full bundle for the browser:
```
gulp browser
```
>This will generate files named bitcore-lib.js and bitcore-lib.min.js.

Reply:
ok the following cause no errors:

```
node: {
  crypto: "empty"
}

node: {
  crypto: true
}

node: {
  crypto: "mock"
}
// this throws an error
```
It looks like the error is might be from webpack it tries to require the module in the bundle step and needs to know what to do here. empty is the most lightweight option.

Reply:
If we want to ensure the crypto polyfill isn't used, then set it to false.  jsonld-signatures is pulling in the bitcore junk.  It's not used, so this just needs to fix it to not explode needlessly.  It seems that "empty" thing happens to work?  But it's applying that globally (I think).  I'm unsure if there may be some unwated side effect of that?  Some feature detection code may be trickier than our tricks.  The other approach that we've used elsewhere is to just target the disabled code directly with this in the webpack (or karma webpack) config:
```js
  node: {
    crypto: false
    ...
  },
  externals: {
    'bitcore-message': '\'bitcore-message\''
  }
```
That solution makes sure crypto is null everywhere, and the bitcoin nonsense is basically compiled out (it pretends it's an external which doesn't really exist, so would fail if code path called).

Reply:
And I should note the long term solution is we'll split up jsonld-signatures.  These short term hacks should be as simple as possible so they are easy to remove later.

Reply:
@davidlehn 

> false: Provide nothing. Code that expects this object may crash with a ReferenceError. Code that attempts to import the module using require('modulename') may trigger a Cannot find module "modulename" error.

false means the tests have an error.
maybe all we need then is

```
resolve: {
  alias: {
    crypto: {}
  }
}
```
hence if a require('crypto') is made it returns an empty object in the tests.